scipy/interpolate/_rgi.py

```diff
@@ -217,20 +217,41 @@ def __init__(self, points, values, method="linear", bounds_error=True,
             self._validate_grid_dimensions(points, method)
         self.method = method
         self.bounds_error = bounds_error
+        # todo: sort checks in order of complexity and runtime, raising early
+        values = self._check_values(values)
+        self._check_dimensionality(points, values)
+        self._check_fill_value(values, fill_value)
+        self.fill_value = fill_value
+        flip = self._check_points(points)
+        self.flip = flip
```

Comment:
`self.flip` should probably be private, i.e. should have a leading underscore. We are trying to minimize the API surface. Also this variable, or the function which computes it, needs a comment for future readers.

Reply:
Renamed to a more descriptive name and made private. Would that be enough? If not where should I put that comment? Above the call to _check_points or inside that function?